soc: imxrt: mimxrt1011 i2s clock fix

imran-sajjad opened the pull request:
Added fix for compiling i2s drivers on the NXP mimxrt1010_evk board.
 
The particular board does not have this peripheral (sai2 in devicetree) and in the header file included for it, does not have these defines: kCLOCK_Sai2Mux, kCLOCK_Sai2PreDiv, kCLOCK_Sai2Div. This change only includes code if the sai2 node exists in the devicetree and is okay. Have not tested with other nxp boards but I think it should work.

Previously created a PR [here](https://github.com/zephyrproject-rtos/hal_nxp/pull/569) but thanks @lucien-nxp, @ZhaoxiangJin for pointing me in the right direction.

butok commented on soc/nxp/imxrt/imxrt10xx/soc.c:
```diff
@@ -320,11 +320,13 @@ void imxrt_audio_codec_pll_init(uint32_t clock_name, uint32_t clk_src,
 		CLOCK_SetDiv(kCLOCK_Sai1Div, clk_src_div);
 		break;
+#if DT_NODE_HAS_STATUS_OKAY(DT_NODELABEL(sai2))
 	case IMX_CCM_SAI2_CLK:
```
Just wondering if we should check sai1 and sai3 as well?

butok replied:
> It looks like sai1 and sai3 exist on every board in this family so the values will always be defined. It's an easy change though and if you think it would still be good then I can just do it

IMHO: Even if they are present/enabled, they can be disabled in an application overlay.

imran-sajjad replied:
Done. Checked similar setup with zephyr v3.7.1 and other mimxrt10xx boards and it compiles.

imran-sajjad committed:
soc: imxrt: mimxrt1011 i2s clock fix

Fix for compiling i2s drivers on the NXP mimxrt1010_evk board.
For mimxrt1011, the defines kCLOCK_Sai2... are not defined as the sai2
peripheral does not exist. Trying to compile gives error. Fixed by adding
check for device tree node around code that uses the defines. Also added
same for sai1 and sai3. Thanks @lucien-nxp, @ZhaoxiangJin from NXP.

Signed-off-by: Imran Sajjad <imran.sajjad@iconfitness.com>
